core/lib/vm/src/oracles/storage.rs

```diff
@@ -88,12 +90,30 @@ impl<S: WriteStorage, H: HistoryMode> StorageOracle<S, H> {
             || *key == storage_key_for_eth_balance(&BOOTLOADER_ADDRESS)
     }
 
+    fn get_initial_value(&self, storage_key: &StorageKey) -> Option<U256> {
+        self.initial_values.inner().get(storage_key).copied()
+    }
+
+    fn set_initial_value(&mut self, storage_key: &StorageKey, value: U256, timestamp: Timestamp) {
+        if self.get_initial_value(storage_key).is_none() {
```

Comment:
You can use `insert` function for `HistoryRecorder<HashMap<T>>`

For an example you can search the following snippet:

```
if !self.initial_values.inner().contains_key(&key) {
            self.initial_values
                .insert(key, current_value, query.timestamp);
        }
```

and replace the snippet with `set_initial_value`